docs/docs/user-guide.md

```diff
@@ -4,115 +4,142 @@ sidebar_position: 2
+nullness of all symbols.
 
-There are two JSpecify annotations that indicate these properties:
+*   [`@Nullable`] applied to a type means a value of that type can be `null`.
+    `@Nullable String x` means that `x` might be `null`.
 
-*   `@Nullable` applied to a type means a reference of that type can be null.
+*   [`@NonNull`] applied to a type means a value of that type cannot be `null`.
+    `@NonNull String x` means that `x` is never `null` (or you have a bug in
+    your code).
 
-*   `@NullMarked` applied to a module, package, or class means that a reference
-    in that scope can't be null unless its type is explicitly marked
+*   [`@NullMarked`] applied to a module, package, class, or method means that a
+    value in that scope can't be `null` unless its type is explicitly marked
     `@Nullable`. (Below we will see that there are some exceptions to this for
     [local variables](#local-variables) and
-    [type variables](#defining-generics).)
+    [type variables](#declaring-generics).) In code covered by `@NullMarked`,
+    `String x` means the same as `@NonNull String x`.
 
-The notion of "can't be null" should really be read with a footnote that says
+*   [`@NullUnmarked`] applied to a package, class, or method undoes the effects
+    of any surrounding `@NullMarked`. Values in its scope generally have
+    unspecified nullness unless they are annotated with `@Nullable` or
+    `@NonNull`, as if there were no enclosing `@NullMarked` at all. A
+    `@NullUnmarked` span may in turn contain nested `@NullMarked` spans to
+    reapply those rules.
+
+The notion of "can't be `null`" should really be read with a footnote that says
 "if all the code in question is `@NullMarked`". For example, if you have some
 code that is not `@NullMarked` and that calls a `@NullMarked` method, then tools
-might allow it to pass a possibly-null value to a method that is expecting a
-"can't be null" parameter.
+might allow it to pass a possibly-`null` value to a method that is expecting a
+"can't be `null`" parameter.
+
+## `@Nullable` and `@NonNull`
 
-## `@Nullable`
+When a type is annotated with [`@Nullable`], it means that a value of the type
+can be `null`. Code that uses those values must be able to deal with the `null`
+case, and it's okay to assign `null` to such variables or pass `null` to those
+parameters.
 
-The `@Nullable` annotation applied to a type means that that use of the type
-includes references that can be null. Code that deals with those references must
-be able to deal with the null case.
+When a type is annotated with [`@NonNull`], it means that no value of the type
+can be `null` (unless there's a bug in your code). Code that uses those values
+can assume they're not `null`, but it's a bad idea to assign `null` to those
+values or pass `null` to those parameters.
 
 ```java
-static void print(@Nullable String x) {
-  System.out.println(String.valueOf(x));
+static @Nullable String emptyToNull(@NonNull String x) {
```

Comment:
I understand the value of being explicit and in starting with the simple, "local" annotations, but I would aggressively, preemptively tell readers that `@NonNull` is not usually what they want, or (if we keep the section about `@NullMarked` above here) I'd use `@NullMarked`.